test_entropy.py

```diff
@@ -20,71 +20,79 @@
+    expected = '71356cdf2f6851c0499b47fe16ef121883f8c58e7d6fb4c33a9017df71f3b0fe21f92d043ee4f19676384a3d943554904caca131269dcb84151ecb2f7ca31902'
+    assert test.hex() == expected
+
+    # with password
+    test = e.bip39_mnemonic_to_entropy("m/83696968'/0'/0'", mnemonic, 'TREZOR')
+    expected = '5c56a2a19ebe8f86c4cbb788dd264bd96387dac2047dac799c51fb6218da0513da44bc0f4603815cc8c8c456dbd3aae79e334fb19dbeffc43fc236d58368ebdb'
+    assert test.hex() == expected
+
+def test_xprv_to_entropy():
+    e = bipentropy.BIPEntropy()
+    test = e.bip32_xprv_to_entropy("m/83696968'/0'/0'", XPRV)
+    expected = '71356cdf2f6851c0499b47fe16ef121883f8c58e7d6fb4c33a9017df71f3b0fe21f92d043ee4f19676384a3d943554904caca131269dcb84151ecb2f7ca31902'
+    assert test.hex() == expected
+
+
+def test_entropy_to_mnemonic():
+    e = bipentropy.BIPEntropy()
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/0'/0'", XPRV)
+
+    words12 = 'illness problem daughter gain lunar then chapter happy wrap resist setup corn'
+    assert e.entropy_to_bip39(entropy, 12) == words12
+
+    words15 = 'illness problem daughter gain lunar then chapter happy wrap resist setup country display glare debate'
+    assert e.entropy_to_bip39(entropy, 15) == words15
+
+    words24 = 'illness problem daughter gain lunar then chapter happy wrap resist setup country display glare delay pupil regular border piano cook warfare what sentence supreme'
+    assert e.entropy_to_bip39(entropy, 24) == words24
+
+def test_wif_from_entropy():
+    # PDG: not sure about this?
+    e = bipentropy.BIPEntropy()
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/0'/0'", XPRV)
+    assert e.entropy_to_wif(entropy) == 'L11mqGbaozsMYDHS7dfZ2bPGL2viSH6zHr69MKwvpxuw7cCR4M1u'
+
+def test_applications():
+    e = bipentropy.BIPEntropy()
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/39'/0'/12'/0'", XPRV)
+    assert entropy[:16].hex() == 'f0337580e36fd50ef8734cd9dcfb9a78'
+    assert e.entropy_to_bip39(entropy, 12) == \
+                     'usual option gasp short wool manual tide hat supply treat track valve'
+
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/39'/0'/18'/0'", XPRV)
+    assert entropy[:24].hex() == '60529dbbf2707ea89e4cd41f7e26fcebf2b492b9a99c5e95'
+    assert e.entropy_to_bip39(entropy, 18) == \
+                     'gate neutral humble top among february junior once buyer van sand subject clip enable trade crime future protect'
+
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/39'/0'/24'/0'", XPRV)
+    assert entropy[:32].hex() == '5166983339d6fc685abe49162327ac2e915fcc17132dad7a2b1e8f324b2f06bd'
+    assert e.entropy_to_bip39(entropy, 24) == \
+                     'fabric crumble art inhale hurt crouch helmet since bike bomb twelve frog bicycle toward fox grant pulp spend sibling bunker caution nurse brain prison'
+
+def test_xprv_application():
+    e = bipentropy.BIPEntropy()
+    result = e.bip32_xprv_to_xprv(0, XPRV)
```

Comment:
I would definitely prefer the library does not use fixed paths. The underlying is reason for these particular functions is the transformation, and if I want to use an arbitrary path, I shouldn't have to rewrite the underlying function to do so. You can always write a wrapper API layer around this.

Reply:
IMHO, the purpose of demo code like this is to fix the paths. It's the embodiment of the BIP.